pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -239,10 +242,15 @@ def solve(self, model, **kwds):
                     from_list=solve_data.best_solution_found.MindtPy_utils.variable_list,
                     to_list=MindtPy.variable_list,
                     config=config)
+                # MindtPy.variable_list does not include the unused variables, but component_data_objects(Var) does. component_data_objects(Var) should be used for both working_model and original_model.
```

Comment:
Why do you need the unused variables? If they're unused in the original model, you're never going to need to do anything with them, right?

Reply:
Yep. Loading the optimal solution has nothing to do with the unused variables. In the working_model, the unused variables have already been filtered out in the variable_list. However, the original_model has not. I am not sure if we need to take efforts to check the unused variables in the original model here. Or just load the optimal solution to both used and unused variables.

Reply:
OK, this makes sense. Then yes: I think you should never use `component_data_objects` to collect the Vars. Then you'll never see unused variables, which is fine because it will make mindtpy match the behavior of other solvers. That is, unused variables are never sent to the solver, and no solution is filled in for them after the solve. (This means unused variables will be stale after the solve, which they should be.)